ansible_base/rbac/api/permissions.py

```diff
@@ -0,0 +1,110 @@
+            return False
+        if request.method in SAFE_METHODS:
+            return has_super_permission(request.user, 'view')
+        return has_super_permission(request.user)
+
+
+class AuthenticatedReadAdminChange(IsSystemAdminOrAuditor):
+    "Any authenticated user can view, but only admin users can do CRUD"
+
+    def has_permission(self, request, view):
+        if not (request.user and request.user.is_authenticated):
+            return False
+        if request.method in SAFE_METHODS:
+            return True
+        return has_super_permission(request.user)
+
+
+class AnsibleBaseObjectPermissions(DjangoObjectPermissions):
+
+    def has_permission(self, request, view):
+        "Some of this comes from ModelAccessPermission. We assume user.permissions is unused"
+        if not request.user or (not request.user.is_authenticated and self.authenticated_users_only):
+            return False
+
+        # Workaround to ensure DjangoModelPermissions are not applied
+        # to the root view when using DefaultRouter.
+        if getattr(view, '_ignore_model_permissions', False):
+            return True
+
+        # Following is DAB RBAC specific, handle add permission checking
+        if request.method == 'POST' and view.action == 'create':
+            # Identify cloned requests used by API browser
+            # the renderer class calls show_form_for_method which checks POST permissions
+            # on a GET request about whether to show the POST form
+            # philosophy is that all users have the abstract ability to post
+            # and the object permissions (data) tells whether it is possible
+            if request.method != request._request.method and (not request.data):
+                return True
+            queryset = self._queryset(view)
+            model_cls = queryset.model
+            parent_field_name = permission_registry.get_parent_fd_name(model_cls)
+            if parent_field_name not in request.data or parent_field_name is None:
+                return bool(request.user.is_superuser)
+            elif parent_field_name not in request.data:
+                return False
+
+            parent_model = permission_registry.get_parent_model(model_cls)
+            parent_obj = parent_model.objects.get(pk=request.data[parent_field_name])
+            return request.user.has_obj_perm(parent_obj, f'add_{model_cls._meta.model_name}')
+
+        # We are not checking many things here, a GET to list views can return 0 objects
+        return True
+
+    def get_required_object_permissions(self, method, model_cls, view=None):
+        special_action = getattr(view, 'rbac_action', None)
+        if special_action:
+            return [f'{special_action}_{model_cls._meta.model_name}']
+        perms = super().get_required_object_permissions(method, model_cls)
+        # Remove add permissions, since they are handled in has_permission
+        return [p for p in perms if 'add_' not in p]
```

Comment:
`.startswith()`.  This is another substring match potential bug, albeit one less likely to fire off than the first one I pointed out.

Reply:
I introduced an entire `is_add_perm` method, so that we collect all the cases where this determination happens.